File: AssetManager/UserInterface/Forms/AssetManager/UpdateTypeForm.vb
Option Explicit On
Imports System.ComponentModel

Public Class UpdateDev

    Public ReadOnly Property UpdateInfo As DeviceUpdateInfoStruct
        Get
            Return NewUpdateInfo
        End Get
    End Property

    Private NewUpdateInfo As DeviceUpdateInfoStruct

    Sub New(parentForm As ExtendedForm, Optional isNoteOnly As Boolean = False)
        InitializeComponent()
        Me.ParentForm = parentForm
        FillComboBox(DeviceAttribute.ChangeType, UpdateTypeCombo)
        If isNoteOnly Then
            UpdateTypeCombo.SelectedIndex = GetComboIndexFromCode(DeviceAttribute.ChangeType, "NOTE")
            UpdateTypeCombo.Enabled = False
            ValidateUpdateType()
        Else
            UpdateTypeCombo.SelectedIndex = -1
        End If
        ShowDialog(parentForm)
    End Sub

    Private Sub SubmitButton_Click(sender As Object, e As EventArgs) Handles SubmitButton.Click
        NewUpdateInfo.Note = Trim(NotesTextBox.Rtf)
        NewUpdateInfo.ChangeType = GetDBValue(DeviceAttribute.ChangeType, UpdateTypeCombo.SelectedIndex)
        NotesTextBox.Text = ""
        UpdateTypeCombo.Enabled = True
        Me.DialogResult = DialogResult.OK
    End Sub

    Private Sub CancelButton_Click(sender As Object, e As EventArgs) Handles CancelButton.Click
        Me.DialogResult = DialogResult.Cancel
    End Sub

    Private Function ValidateUpdateType() As Boolean
        If UpdateTypeCombo.SelectedIndex > 0 Then
            ErrorProvider1.SetError(UpdateTypeCombo, "")
            SubmitButton.Enabled = True
            Return True
        Else
            SubmitButton.Enabled = False
            UpdateTypeCombo.Focus()
            ErrorProvider1.SetError(UpdateTypeCombo, "Please select a change type.")
        End If
        Return False
    End Function

    Private Sub UpdateTypeCombo_ChangeType_Validating(sender As Object, e As CancelEventArgs) Handles UpdateTypeCombo.Validating
        e.Cancel = Not ValidateUpdateType()
    End Sub

    Private Sub UpdateTypeCombo_SelectionChangeCommitted(sender As Object, e As EventArgs) Handles UpdateTypeCombo.SelectionChangeCommitted
        ValidateUpdateType()
    End Sub
End Class

File: AssetManager.ConvertedToC#/UserInterface/Forms/AssetManager/UpdateTypeForm.cs
using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Drawing;
using System.Diagnostics;
using System.Windows.Forms;
using System.Linq;
using System.Threading.Tasks;
using System.ComponentModel;
using AssetManager.UserInterface.CustomControls;

namespace AssetManager.UserInterface.Forms.AssetManager
{

	public partial class UpdateDev : ExtendedForm
	{

		public DeviceUpdateInfoStruct UpdateInfo {
			get { return NewUpdateInfo; }
		}


		private DeviceUpdateInfoStruct NewUpdateInfo;
		public UpdateDev(ExtendedForm parentForm, bool isNoteOnly = false)
		{
			InitializeComponent();
			this.ParentForm = parentForm;
			AttribIndexFunctions.FillComboBox(GlobalInstances.DeviceAttribute.ChangeType,  UpdateTypeCombo);
			if (isNoteOnly) {
				UpdateTypeCombo.SelectedIndex = AttribIndexFunctions.GetComboIndexFromCode(GlobalInstances.DeviceAttribute.ChangeType, "NOTE");
				UpdateTypeCombo.Enabled = false;
				ValidateUpdateType();
			} else {
				UpdateTypeCombo.SelectedIndex = -1;
			}
			ShowDialog(parentForm);
		}

		private void SubmitButton_Click(object sender, EventArgs e)
		{
			NewUpdateInfo.Note = Strings.Trim(NotesTextBox.Rtf);
			NewUpdateInfo.ChangeType = AttribIndexFunctions.GetDBValue(GlobalInstances.DeviceAttribute.ChangeType, UpdateTypeCombo.SelectedIndex);
			NotesTextBox.Text = "";
			UpdateTypeCombo.Enabled = true;
			this.DialogResult = DialogResult.OK;
		}

		private void CancelButton_Click(object sender, EventArgs e)
		{
			this.DialogResult = DialogResult.Cancel;
		}

		private bool ValidateUpdateType()
		{
			if (UpdateTypeCombo.SelectedIndex > 0) {
				ErrorProvider1.SetError(UpdateTypeCombo, "");
				SubmitButton.Enabled = true;
				return true;
			} else {
				SubmitButton.Enabled = false;
				UpdateTypeCombo.Focus();
				ErrorProvider1.SetError(UpdateTypeCombo, "Please select a change type.");
			}
			return false;
		}

		private void UpdateTypeCombo_ChangeType_Validating(object sender, CancelEventArgs e)
		{
			e.Cancel = !ValidateUpdateType();
		}

		private void UpdateTypeCombo_SelectionChangeCommitted(object sender, EventArgs e)
		{
			ValidateUpdateType();
		}

	}
}
